lib/gourami/extensions/resources.rb

```diff
@@ -91,6 +91,28 @@ def any_resource_errors?
         resource_errors.values.flat_map(&:values).map(&:values).flatten.any?
       end
 
+      # Replace the existing resource errors with the provided errors Hash.
+      #
+      # @param new_resource_errors [Hash<Symbol, Hash<Symbol, Hash<Symbol, Array>>>]
+      #
+      # @return [Hash<Symbol, Hash<Symbol, Hash<Symbol, Array>>>]
+      def clear_and_set_resource_errors(new_resource_errors)
+        new_resource_errors = new_resource_errors.dup
+        resource_errors.clear
+        resource_errors.merge!(new_resource_errors)
+
+        resource_errors
+      end
+
+      def handle_validation_error(error)
+        super(error)
+        clear_and_set_resource_errors(error.resource_errors) unless error.resource_errors.nil?
```

Comment:
`unless error.resource_errors.nil?` What's that for?

Reply:
This method could be called without any resource errors right?